apps/dotcdn/src/app/app.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DotCDNState } from './app.interface';
+import { ChartPeriod } from './app.enums';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { DotCDNStore } from './dotcdn.component.store';
+import { Observable } from 'rxjs';
+import { SelectItem } from 'primeng/api';
+import { ChartOptions } from 'chart.js';
+import { take } from 'rxjs/operators';
+
+@Component({
+    selector: 'dotcms-root',
+    templateUrl: './app.component.html',
+    styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit {
+    @ViewChild('chart', { static: true }) chart: any;
+    purgeZoneForm: FormGroup;
+    periodValues: SelectItem[] = [
+        { label: 'Last 30 days', value: ChartPeriod.Last30Days },
+        { label: 'Last 60 days', value: ChartPeriod.Last60Days }
+    ];
+
+    selectedPeriod: SelectItem<string> = { value: ChartPeriod.Last30Days };
+    vm$: Observable<Partial<DotCDNState>> = this.dotCdnStore.vm$;
+    vmPurgeLoaders$: Observable<Partial<DotCDNState>> = this.dotCdnStore.vmPurgeLoaders$;
+    chartHeight = '25rem';
+    options: ChartOptions;
+
+    constructor(private fb: FormBuilder, private dotCdnStore: DotCDNStore) {}
+
+    ngOnInit(): void {
+        this.setChartOptions();
+        this.purgeZoneForm = this.fb.group({
+            purgeUrlsTextArea: ''
+        });
+    }
+
+    /**
+     *  Handles the period change
+     *
+     * @param {*} event
+     * @memberof AppComponent
+     */
+    changePeriod(element: HTMLTextAreaElement): void {
+        this.dotCdnStore.getChartStats(element.value);
+    }
+
+    /**
+     * Purges the entire cache
+     *
+     * @memberof AppComponent
+     */
+    purgePullZone(): void {
+        const invalidateAll = true;
+        this.dotCdnStore.purgeCDNCache(invalidateAll);
```

Comment:
```suggestion
        this.dotCdnStore.purgeCDNCacheAll();
```

Using a param as flag is bad practice because you are making you method to do more than one thing.